model/internal/cmd/pdatagen/internal/base_fields.go

```diff
@@ -64,10 +64,17 @@ const oneOfTypeAccessorHeaderTestTemplate = `func Test${structName}${originField
 	assert.Equal(t, "", ${typeName}(1000).String())`
 
 const accessorsOneOfMessageTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
-// Calling this function when ${originOneOfFieldName}Type() != ${typeName} will cause a panic.
+//
+// Calling this function when ${originOneOfFieldName}Type() != ${typeName} returns an invalid 
+// zero-initiazed instance of ${returnType}. Note that using such ${returnType} instance can cause panic.
+//
 // Calling this function on zero-initialized ${structName} will cause a panic.
 func (ms ${structName}) ${fieldName}() ${returnType} {
-	return new${returnType}((*ms.orig).${originOneOfFieldName}.(*${originStructType}).${originFieldName})
+	v := ms.orig.Get${originFieldName}()
+	if v == nil {
+		return ${returnType}{}
+	}
+	return new${returnType}(v)
```

Comment:
nit: This has an extra "if check" in the get. Try to do the type conversion here and use ok directly.

Reply:
done